pkg/operator/mon/mon.go

```diff
@@ -128,44 +128,50 @@ func (c *Cluster) Start() error {
 	}
 
 	if len(c.clusterInfo.Monitors) < c.Size {
-		// init the mons config
-		mons := c.initMonConfig(c.Size)
-
-		// Assign the pods to nodes
-		if err := c.assignMons(mons); err != nil {
-			return fmt.Errorf("failed to assign pods to mons. %+v", err)
+		c.startMons()
+	} else {
+		// Check the health of a previously started cluster
+		if err := c.checkHealth(); err != nil {
+			logger.Warningf("failed to check mon health %+v. %+v", c.clusterInfo.Monitors, err)
 		}
+	}
 
-		// Start one monitor at a time
-		for i := len(c.clusterInfo.Monitors); i < c.Size; i++ {
-			// Init the mon IPs
-			if err := c.initMonIPs(mons[0 : i+1]); err != nil {
-				return fmt.Errorf("failed to init mon services. %+v", err)
-			}
+	return nil
+}
 
-			// save the mon config after we have "initiated the IPs"
-			if err := c.saveMonConfig(); err != nil {
-				return fmt.Errorf("failed to save mons. %+v", err)
-			}
+func (c *Cluster) startMons() error {
```

Comment:
was there anything changed in this method? it looks like it was just refactored. Looks good, just wanted to confirm if i missed a change

Reply:
I just moved the logic to an own function so there is no need to duplicated.